crates/pop-parachains/templates/pallet/src/config_preludes.rs.templ

```diff
@@ -0,0 +1,139 @@
+    #[inject_runtime_type]
+    type RuntimeHoldReason = ();
+    #[inject_runtime_type]
+    type RuntimeFreezeReason = ();
+    {{- endif -}}
+    {{- for config_type in pallet_config_types -}}
+    {{- match config_type.1 -}}
+    {{- when TemplatePalletConfigTypesDefault::Default }}
+    type ^^config_type.2^^ = todo!();
+    {{- when TemplatePalletConfigTypesDefault::NoDefaultBounds }}
+    type ^^config_type.2^^ = todo!();
+    {{- else -}}
+    {{- endmatch -}}
+    {{- endfor }}
+}
+
+/// Default configurations of this pallet in a solochain environment.
+pub struct SolochainDefaultConfig;
+
+#[derive_impl(frame_system::config_preludes::SolochainDefaultConfig, no_aggregated_types)]
+impl frame::deps::frame_system::DefaultConfig for SolochainDefaultConfig {}
+
+#[register_default_impl(SolochainDefaultConfig)]
+impl DefaultConfig for SolochainDefaultConfig {
+    {{- if using_runtime_event -}}
+    #[inject_runtime_type]
+    type RuntimeEvent = ();
+    {{- endif -}}
+    {{- if using_runtime_origin -}}
+    #[inject_runtime_type]
+    type RuntimeOrigin = ();
+    {{- endif -}}
+    {{- if using_currency -}}
+    #[inject_runtime_type]
+    type RuntimeHoldReason = ();
+    #[inject_runtime_type]
+    type RuntimeFreezeReason = ();
+    {{- endif -}}
+    {{- for config_type in pallet_config_types -}}
+    {{- match config_type.1 -}}
+    {{- when TemplatePalletConfigTypesDefault::Default }}
+    type ^^config_type.2^^ = todo!();
+    {{- when TemplatePalletConfigTypesDefault::NoDefaultBounds }}
+    type ^^config_type.2^^ = todo!();
+    {{- else -}}
+    {{- endmatch -}}
+    {{- endfor }}
+}
+
+/// Default configurations of this pallet in a solochain environment.
+pub struct RelayChainDefaultConfig;
+
+#[derive_impl(frame_system::config_preludes::RelayChainDefaultConfig, no_aggregated_types)]
+impl frame::deps::frame_system::DefaultConfig for RelayChainDefaultConfig {}
+
+#[register_default_impl(RelayChainDefaultConfig)]
+impl DefaultConfig for RelayChainDefaultConfig {
+    {{- if using_runtime_event -}}
+    #[inject_runtime_type]
+    type RuntimeEvent = ();
+    {{- endif -}}
+    {{- if using_runtime_origin -}}
+    #[inject_runtime_type]
+    type RuntimeOrigin = ();
+    {{- endif -}}
+    {{- if using_currency -}}
+    #[inject_runtime_type]
+    type RuntimeHoldReason = ();
+    #[inject_runtime_type]
+    type RuntimeFreezeReason = ();
+    {{- endif -}}
+    {{- for config_type in pallet_config_types -}}
+    {{- match config_type.1 -}}
+    {{- when TemplatePalletConfigTypesDefault::Default }}
+    type ^^config_type.2^^ = todo!();
+    {{- when TemplatePalletConfigTypesDefault::NoDefaultBounds }}
+    type ^^config_type.2^^ = todo!();
+    {{- else -}}
+    {{- endmatch -}}
+    {{- endfor }}
+} 
+
+/// Default configurations of this pallet in a solochain environment.
+pub struct ParaChainDefaultConfig;
+
+#[derive_impl(frame_system::config_preludes::ParaChainDefaultConfig, no_aggregated_types)]
```

Comment:
```suggestion
#[derive_impl(frame_system::config_preludes::ParachainDefaultConfig, no_aggregated_types)]
```

Reply:
This wouldn't work right now, frame system's config_preludes module defines `ParaChainDefaultConfig`:![frame_system::ParaChainDefaultConfig](https://github.com/tsenovilla/polkadot-sdk/blob/master/substrate/frame/system/src/lib.rs#L343) . And yeah, it also defines `SolochainDefaultConfig`:
![frame_system::SolochainDefaultConfig](https://github.com/tsenovilla/polkadot-sdk/blob/master/substrate/frame/system/src/lib.rs#L343).

It's a bit confusing, but we have to derive in the same way it's defined in the polkadot-sdk, so it's better to keep the same struct name in the template generated config_prelude IMO